src/ORM/Association.php

```diff
@@ -1014,7 +1014,7 @@ protected function _formatAssociationResults(Query $query, Query $surrogate, arr
             }
             $extracted = new Collection($extracted);
             foreach ($formatters as $callable) {
-                $extracted = new ResultSetDecorator($callable($extracted));
+                $extracted = new ResultSetDecorator($callable($extracted, $query));
```

Comment:
This `formatResults()` should now receive `$query` as an argument. We should update to use the new parameter.